control-plane-operator/hostedclusterconfigoperator/controllers/resources/manifests/pullsecret.go

```diff
@@ -20,3 +30,56 @@ func PullSecretTargetNamespaces() []string {
 		"openshift",
 	}
 }
+
+func AdditionalPullSecret() *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "additional-pull-secret",
+			Namespace: GlobalPullSecretNamespace,
+		},
+	}
+}
+
+func GlobalPullSecretDaemonSet() *appsv1.DaemonSet {
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "global-pull-secret-syncer",
+			Namespace: GlobalPullSecretNamespace,
+		},
+	}
+}
+
+func GlobalPullSecret() *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "global-pull-secret",
+			Namespace: GlobalPullSecretNamespace,
+		},
+		Type: corev1.SecretTypeDockerConfigJson,
+	}
+}
+
+func GlobalPullSecretServiceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "global-pull-secret-syncer",
+			Namespace: GlobalPullSecretNamespace,
+		},
+	}
+}
+
+func GlobalPullSecretRole() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "global-pull-secret-syncer",
+		},
+	}
+}
+
+func GlobalPullSecretRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
```

Comment:
this needs to be a roleBinding?

Reply:
Then you need several roles. We don't want to expose an overprivileged SA in all Nodes.